NAS-102494 / 11.3 / Allow lzma compression for importing/exporting jails

sonicaj opened the pull request:
This PR aims to expose option to allow lzma compression for import/export of jails.

sonicaj committed:
Add ability to export jails with lzma compression

This commit adds an option to select compression algorithm for exporting a jail with either zipped or lzma compression.

sonicaj committed:
Expose compression algo in cli for exporting jails

This commit exposes compression algo selection in cli for exporting jails.

sonicaj committed:
Allow importing lzma based exported jails

This commit allows importing jails which were compressed with lzma compression.

sonicaj committed:
Expose compression algo in cli for importing jails

This commit exposes compression algo selection in cli for importing jails.

william-gr commented on iocage_lib/iocage.py:
```diff
@@ -1386,9 +1388,11 @@ def get(
             return jail_list
 
-    def import_(self):
+    def import_(self, compression_algo='zip'):
```
Is there a way to autodetect the file instead?

sonicaj replied:
Well we can but that means an exception if we have 2 exported jails with similar names but different algorithms. I guess i raise an exception in that case and that will allow this to automatically proceed if there are no conflicts. Thoughts ?

sonicaj committed:
Allow automatic discovery of jail image

This commit introduces automatic discovery of jail image irrespective of the algorithm type.

william-gr reviewed (commented):
I find it weird that we cannot provide a path of the image we want to import?

sonicaj commented:
Well we have a defined dataset for it right now where all images must live. But for import cases it indeed is. Would you like me to add an option ?

william-gr commented:
Yes, also seems easier for use case instead of having to move it first to images dataset.

sonicaj committed:
Add support for specifying path for importing jail

This commit adds support for specifying path for importing a jail.
